### PR TITLE
Fix 'return log_error()' and 'return log_warning()' patterns

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

**Comment by keszybz** [comment]
bionic-amd64 failed in test-path with a timeout, I hope that's unrelated.

File: src/core/execute.c [hunk 1]
```diff
@@ -6281,8 +6281,13 @@ int exec_runtime_deserialize_one(Manager *m, const char *value, FDSet *fds) {
...
                 n = strcspn(v, " ");
                 buf = strndupa(v, n);
-                if (safe_atoi(buf, &fdpair[0]) < 0 || !fdset_contains(fds, fdpair[0]))
-                        return log_debug("Unable to process exec-runtime netns fd specification.");
+
+                r = safe_atoi(buf, &fdpair[0]);
+                if (r < 0)
+                        return log_debug_errno(r, "Unable to parse exec-runtime specification netns-socket-0=%s: %m", buf);
+                if (!fdset_contains(fds, fdpair[0]))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EBADFD),
```
Comment (poettering) [inline src/core/execute.c]: i think EBADF would be the better error here actually, i.e. it's not the state of the fd it's the fd itself that is bad here

File: src/core/execute.c [hunk 2]
```diff
@@ -6295,8 +6300,12 @@ int exec_runtime_deserialize_one(Manager *m, const char *value, FDSet *fds) {
 
                 n = strcspn(v, " ");
                 buf = strndupa(v, n);
-                if (safe_atoi(buf, &fdpair[1]) < 0 || !fdset_contains(fds, fdpair[1]))
-                        return log_debug("Unable to process exec-runtime netns fd specification.");
+                r = safe_atoi(buf, &fdpair[1]);
+                if (r < 0)
+                        return log_debug_errno(r, "Unable to parse exec-runtime specification netns-socket-1=%s: %m", buf);
+                if (!fdset_contains(fds, fdpair[0]))
+                        return log_debug_errno(SYNTHETIC_ERRNO(EBADFD),
```
Comment (poettering) [inline src/core/execute.c]: ditto

File: src/core/execute.c [hunk 3]
```diff
@@ -6281,8 +6281,13 @@ int exec_runtime_deserialize_one(Manager *m, const char *value, FDSet *fds) {
 
                 n = strcspn(v, " ");
                 buf = strndupa(v, n);
-                if (safe_atoi(buf, &fdpair[0]) < 0 || !fdset_contains(fds, fdpair[0]))
-                        return log_debug("Unable to process exec-runtime netns fd specification.");
+
+                r = safe_atoi(buf, &fdpair[0]);
+                if (r < 0)
+                        return log_debug_errno(r, "Unable to parse exec-runtime specification netns-socket-0=%s: %m", buf);
```
Comment (poettering) [inline src/core/execute.c:6287]: ouch!

**Review by poettering — CHANGES_REQUESTED** [review]
looks good, just some minor error code nitpicking

File: src/core/execute.c (hunk 1 above)
Comment (keszybz) [reply src/core/execute.c]: Indeed.

**Comment by keszybz** [comment]
Updated with EBADFD→EBADF, no other chagnes.

**Pull request merged**